packages/xen-api/index.mjs

```diff
@@ -302,6 +325,10 @@ export class Xapi extends EventEmitter {
   // RPC calls
   // ===========================================================================
 
+  computeCacheKey(...args) {
+    return stableJson(args)
```

Comment:
what about using a hash of this value ? that way keys will have a fixed length

with a `import {createHash} from crypto` at the start of the file 
```suggestion
    const string = stableJson(args)
    var shasum = createHash('sha1')
    shasum.update(string)
    return shasum.digest('hex') 
```

since the json is computed sequentially, it could be also used to limit memory consumption of bigger json

Reply:
I implemented it based on your suggestion.

I used `sha256` instead of `sha1` to reduce the risk of collisions and `base64` instead of `hex` for shorter keys.